### DISCUSSION TITLE
How to take points in the field?

### DISCUSSION THREAD

**Discussion by Fankunkun123** [lululxvi/deepxde#1597] [Q&A]
Hello, sorry to bother you. I am currently experiencing the following issue when reimplementing the code for deepXDE: After taking points in the domain, how do I extract all the points "train_x" in the domain, boundary, and initial conditions? I hope you can help me answer this question. Thank you!
![微信截图_20231220152707](https://github.com/lululxvi/deepxde/assets/144784075/7f80e4de-d767-48ac-a9c7-177937c95f12)

**Comment by lululxvi** [comment]
Use `data.num_bcs`

https://github.com/lululxvi/deepxde/blob/d342b4d3a43df4642eb8bdf5a14574ea4052d7c7/deepxde/data/pde.py#L64C9-L64C16

**Comment by Fankunkun123** [comment]
Hello, Dr. Lu. First of all, thank you very much for answering my question, which has been solved very well. Thank you. 

> I used the data.train_points() method to retrieve the training points, which is all the training points(including boundary points, initial points and in-domain points).
